docs/guides/high-availability/concepts-threads.adoc

```diff
@@ -0,0 +1,69 @@
+
+{project_name} requests, as well as all probes, are handled by the Quarkus executor pool.
+
+The Quarkus executor thread pool is configured in https://quarkus.io/guides/all-config#quarkus-core_quarkus.thread-pool.max-threads[`quarkus.thread-pool.max-threads`] and has a maximum size of at least 200 threads.
+Depending on the available CPU cores, it can grow even larger.
+Threads are created as needed, and will end when no longer needed, so the system will scale up and down as needed.
+
+When the load and the number of threads increases, the bottleneck will usually be the database connections.
+Once a request cannot acquire a database connection, it will fail with a message in the log like `Unable to acquire JDBC Connection`.
+The caller will receive a response with a 5xx HTTP status code indicating a server side error.
+
+With the number of threads in the executor pool being an order of magnitude larger than the number of database connections and with requests failing when no database connection is available within the https://quarkus.io/guides/all-config#quarkus-agroal_quarkus.datasource.jdbc.acquisition-timeout[`quarkus.datasource.jdbc.acquisition-timeout`] (5 seconds default), this is somewhat of a https://en.wikipedia.org/wiki/Demand_response#Load_shedding[load-shedding behavior] where it returns an error response instead of queueing requests for an indefinite amount of time.
+
+=== JGroups connection pool
+
+The combined number of executor threads in all {project_name} nodes in the cluster should not exceed the number of threads available in JGroups thread pool to avoid the error `org.jgroups.util.ThreadPool: thread pool is full`.
+To see the error the first time it happens, the system property `jgroups.thread_dumps_threshold` needs to be set to `1`, as otherwise the message appears only after 10000 threads have been rejected.
+
+--
+include::partials/threads/executor-jgroups-thread-calculation.adoc[]
+--
+
+Use the metrics `vendor_jgroups_tcp_get_thread_pool_size` to monitor the total JGroup threads in the pool and `vendor_jgroups_tcp_get_thread_pool_size_active` for the threads active in the pool.
+This is useful to monitor that limiting the Quarkus thread pool size keeps the number of active JGroup threads below the maximum JGroup thread pool size.
+
+[#load-shedding]
+=== Load Shedding
+
+By default, {project_name} will queue all incoming requests infinitely, even if the request processing stalls.
+This will use additional memory in the Pod, can exhaust resources in the load balancers, and the requests will eventually time out on the client side without the client knowing if the request has been processed.
+To limit the number of queued requests in {project_name}, set an additional Quarkus configuration option.
+
+Configure `quarkus.thread-pool.queue-size` to specify a maximum queue length to allow for effective load shedding once this queue size is exceeded: {project_name} will return HTTP Status code 500 (server error).
```

Comment:
Our load-shedding mechanism is/will return 503 instead, right? Not sure if `500` is the right status code...